compiler/noirc_evaluator/src/ssa/opt/loop_invariant.rs

```diff
@@ -109,6 +110,28 @@ impl<'f> LoopInvariantContext<'f> {
 
                 if hoist_invariant {
                     self.inserter.push_instruction(instruction_id, pre_header);
+
+                    // We track whether we may mutate MakeArray instructions before we deduplicate
+                    // them but we still need to issue an extra inc_rc in case they're mutated afterward.
+                    if matches!(
+                        self.inserter.function.dfg[instruction_id],
+                        Instruction::MakeArray { .. }
+                    ) {
+                        let results =
+                            self.inserter.function.dfg.instruction_results(instruction_id).to_vec();
+                        let results = vecmap(results, |value| self.inserter.resolve(value));
+                        assert_eq!(
+                            results.len(),
+                            1,
+                            "ICE: We expect only a single result from an `Instruction::MakeArray`"
+                        );
```

Comment:
If we only expect a single result, why allocate the temporary Vec?
Also, can't we get two results in the case of making slices?

Reply:
Slices are constructed using two instructions. One for the array representing the slice and one for the dynamic length.
e.g.
```
v3 = make_array [] : [Field]
    v4 = allocate -> &mut u32
    store u32 0 at v4
```

Reply:
Switched to just access the first element